netmiko/base_connection.py

```diff
@@ -386,10 +388,18 @@ def __init__(
                     no_log=no_log,
                     record_writes=session_log_record_writes,
                 )
+            elif isinstance(session_log, SessionLog):
+                # SessionLog object
+                self.session_log = session_log
+                # Add filter after self.session_log is constructed
+                log.addFilter(SecretsFilter(no_log=self.session_log.no_log))
+                # Check if filename was provided in session_log
+                if self.session_log.file_name is not None:
+                    self.session_log.open()
```

Comment:
I probably wouldn't call `open()`, I would pass in an already open() session_log. I think this would give end-users more flexibility (for example, if they were trying to do something more complex like multiple-thread session_log.

Reply:
Okay. Would you mind suggesting how that should be accomplished? I'm unsure how to implement that... maybe I'm overthinking it.

Reply:
Just remove lines `397` and `398`.

The SessionLog object would already have been `opened` when passed into Netmiko (i.e. so you pass in an already opened `SessionLog` object that you can just `write()` )

Reply:
Ah yeah, I figured I was overthinking it 😅. Changes pushed.